File: examples/example2/shapes/AreaCalculator.java
package example2.shapes;

import java.util.List;

public class AreaCalculator {
    private List<Object> shapes;

    public AreaCalculator(List<Object> shapes) {
        this.shapes = shapes;
    }

    public double sum() {
        double totalArea = 0;

        for (Object shape : shapes) {
            if (shape instanceof Square) {
                Square square = (Square) shape;
                totalArea += Math.pow(square.getLength(), 2);
            } else if (shape instanceof Circle) {
                Circle circle = (Circle) shape;
                totalArea += Math.PI * Math.pow(circle.getRadius(), 2);
            }
        }

        return totalArea;
    }
}

File: examples/example2/shapes/areaCalculator.py
import math
import json

class AreaCalculator:
    def __init__(self, shapes=[]) -> None:
        self.shapes = shapes
    
    def sum(self):
        areas = []
        for shape in self.shapes:
            if isinstance(shape, Square):
                areas.append(math.pow(shape.length, 2))
            elif isinstance(shape, Circle):
                areas.append(math.pi * math.pow(shape.radius, 2))
        return sum(areas)

class Square:
    def __init__(self, length) -> None:
        self.length = length

class Circle:
    def __init__(self, radius) -> None:
        self.radius = radius

class SumCalculator:
    def __init__(self, areaCalculator: AreaCalculator) -> None:
        self.calculator = areaCalculator

    def JSON(self):
        data = {
            'sum': self.calculator.sum()
        }
        return json.dumps(data)

    def HTML(self):
        return f"Sum of the areas of provided shapes: {self.calculator.sum()}"


# Beispiel für die Verwendung:
shapes = [
    Circle(2),
    Square(5),
    Square(6)
]

areas = AreaCalculator(shapes)
output = SumCalculator(areas)

# Ausgabe der Ergebnisse
print(output.JSON())
print(output.HTML())